subxt/src/storage/storage_key.rs

```diff
@@ -186,32 +172,67 @@ impl StorageKey for Vec<scale_value::Value> {
     }
 
     fn decode_from_bytes(
-        cursor: &mut &[u8],
+        bytes: &mut &[u8],
         hashers_and_ty_ids: &mut &[(StorageHasher, u32)],
         metadata: &Metadata,
     ) -> Result<Self, Error>
     where
         Self: Sized + 'static,
     {
-        let mut hashers_and_ty_ids_iter = hashers_and_ty_ids.iter();
         let mut result: Vec<scale_value::Value> = vec![];
-        let mut n = 0;
-        while !cursor.is_empty() {
-            let Some((hasher, ty_id)) = hashers_and_ty_ids_iter.next() else {
-                // Still bytes left, but no hashers and type ids anymore to pull from: this is an unexpected error.
-                return Err(StorageAddressError::UnexpectedAddressBytes.into());
-            };
-            strip_storage_hash_bytes(cursor, hasher)?;
-            let decoded = DecodedValueThunk::decode_with_metadata(cursor, *ty_id, metadata)?;
-            let value = decoded.to_value()?;
-            result.push(value.remove_context());
-            n += 1;
+        for (hasher, ty_id) in hashers_and_ty_ids.iter() {
```

Comment:
I had a look and `DecodedValueThunk::decode_with_metadata` looks like it consumes _all_ input bytes (not sure why tbh!), so the value decoding would have failed when more than one hasher I think.

I just decoded directly into a Value and figured it may as well go into a for loop where we check for remaining bytes at the end.